pigpen-cascading/src/main/java/pigpen/cascading/SingleIterationSeq.java

```diff
@@ -53,16 +63,24 @@ public Object first() {
   }
 
   public ISeq next() {
-    if (state.isRealized == false) {
+    if (state.rest == state) {
```

Comment:
That's a good point - I guess if they use a `pmap`, we could run into issues. Let's leave it in.